e2e/main.py

```diff
@@ -0,0 +1,82 @@
+import os
+import sys
+import time
+import logging
+import unittest
+import argparse
+
+import docker
+import portpicker
+from docker.models.containers import Container
+
+import pytezos
+
+PROJECT_ROOT = os.path.join(os.path.dirname(__file__), "../")
+
+sys.path.append(os.path.join(PROJECT_ROOT, "client"))
+from checker import *
+
+class SandboxedTestCase(unittest.TestCase):
+    def setUp(self):
+       self.docker_client = docker.from_env()
+
+       port = portpicker.pick_unused_port()
```

Comment:
TIL about this

Reply:
Yeah, it is handy. It's still not great since it doesn't guarantee the port staying empty until we bind to it later on, but should be good enough here.